.envrc

```diff
@@ -1,26 +1,35 @@
-# Build a folder ./.env that contains all the tools
+# Build a folder ./.env that contains the dev-shell which load all the right env
+# vars.
+#
 # To speed up the nix evaluation, we only rebuild our environment when `./nix`
-# changes. We do this by adding its contents to the nix store and using its
-# store path as a cache key. We only use this to build the dev environment. This
-# will fail building in case anything inside the devEnv is reaching outside
-# the nix/ subfolder
-store_path=$(nix-store --add ./nix/)
+# or any of the `default.nix` files change. We do this by adding all these files
+# to the nix store and using the store paths as a cache key.
+
+store_paths=$(find . -name default.nix  | grep -v '^./nix' | grep -v '^./dist-newstyle' | xargs nix-store --add ./nix)
+
+# store_paths=$(nix-store --add ./nix/)
 layout_dir=$(direnv_layout_dir)
+env_dir=./.env
 [[ -d "$layout_dir" ]] || mkdir -p "$layout_dir"
-if [[ ! -d ./.env || ! -f "$layout_dir/nix-rebuild" || "$store_path" != $(< "$layout_dir/nix-rebuild" ) ]]; then
 
-  # this builds "devEnv" defined in ./nix/default.nix and outputs to ./.env
-  nix-build "$store_path" -A devEnv --out-link ./.env
+if [[ ! -d "$env_dir" || ! -f "$layout_dir/nix-rebuild" || "$store_paths" != $(< "$layout_dir/nix-rebuild" ) ]]; then
+
+  [[ -d "$env_dir" ]] || mkdir -p "$env_dir"
+
+  # this builds "wireServer.devShell" defined in ./nix/default.nix and outputs to ./.env
+  nix print-dev-env --impure --expr "(import ./nix).wireServer.devShell" > "${env_dir}/dev-shell.sh"
```

Comment:
This fails for me:

```
❯ direnv allow
direnv: loading ~/dev/wire/wire-server/.envrc                                                            
./.envrc:20: ./.env/dev-shell.sh: Read-only file system
this derivation will be built:
  /nix/store/4acqazyg0sd2q2si8k9pdy5gdkkmz6xl-profile-env.drv
this path will be fetched (0.01 MiB download, 0.04 MiB unpacked):
  /nix/store/xyksslzlz3m0grffb2qy4jqgg8k7pkvb-stdenv-linux
copying path '/nix/store/xyksslzlz3m0grffb2qy4jqgg8k7pkvb-stdenv-linux' from 'https://cache.nixos.org'...
building '/nix/store/4acqazyg0sd2q2si8k9pdy5gdkkmz6xl-profile-env.drv'...
error: creating symlink from '/home/flokli/dev/wire/wire-server/.env/profile-env.tmp-1020538-1297609385' to '/nix/store/8sdg1skjhc74x379gx3hld656rg5kjy6-profile-env': Read-only file system
./.envrc:27: ./.env/dev-shell.sh: No such file or directory
direnv: export +LC_ALL
```

Reply:
Ah, you probably have the old `.env` lying around. I added some code to delete the symlink if it exists.